devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy

```diff
@@ -0,0 +1,683 @@
+}
+
+def executeConfigure() {
+	def cmds = [
+			sensorBinaryGetCmd(),
+			batteryGetCmd(),
+			sensorMultilevelGetCmd(tempSensorType),
+			sensorMultilevelGetCmd(lightSensorType)
+	]
+
+	cmds += getConfigCmds()
+
+	sendCommands(delayBetween(cmds, 500))
+}
+
+private getConfigCmds() {
+	def cmds = []
+	configParams.each { param ->
+		def storedVal = getParamStoredValue(param.num)
+		if (state.refreshConfig) {
+			cmds << configGetCmd(param)
+		}
+		else if ("${storedVal}" != "${param.value}") {
+			def paramVal = param.value
+			logDebug "Changing ${param.name}(#${param.num}) from ${storedVal} to ${paramVal}"
+			cmds << configSetCmd(param, paramVal)
+			cmds << configGetCmd(param)
+
+			if (param.num == minTemperatureOffsetParam.num) {
+				cmds << "delay 3000"
+				cmds << sensorMultilevelGetCmd(tempSensorType)
+			}
+			else if (param.num == minHumidityOffsetParam.num) {
+				cmds << "delay 3000"
+				cmds << sensorMultilevelGetCmd(lightSensorType)
+			}
+		}
+	}
+	state.refreshConfig = false
+	return cmds
+}
+
+private sendCommands(cmds) {
+	def actions = []
+	cmds?.each {
+		actions << new physicalgraph.device.HubAction(it)
+	}
+	sendHubCommand(actions, 100)
+	return []
+}
+
+
+// Required for HealthCheck Capability, but doesn't actually do anything because this device sleeps.
+def ping() {
+	logDebug "ping()"
+}
+
+
+// Forces the configuration to be resent to the device the next time it wakes up.
+def refresh() {
+	logForceWakeupMessage "The sensor data will be refreshed the next time the device wakes up."
+	state.lastBattery = null
+	if (!state.refreshSensors) {
+		state.refreshSensors = true
+	}
+	else {
+		state.refreshConfig = true
+	}
+	refreshPendingChanges()
+	return []
+}
+
+private logForceWakeupMessage(msg) {
+	logDebug "${msg}  You can force the device to wake up immediately by holding the z-button for 2 seconds."
+}
+
+
+def parse(String description) {
+	def result = []
+	try {
+		def cmd = zwave.parse(description, commandClassVersions)
+		if (cmd) {
+			result += zwaveEvent(cmd)
+		}
+		else {
+			logDebug "Unable to parse description: $description"
+		}
+
+		sendEvent(name: "lastCheckIn", value: convertToLocalTimeString(new Date()), displayed: false)
+	}
+	catch (e) {
+		log.error "$e"
+	}
+	return result
+}
+
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapCmd = cmd.encapsulatedCommand(commandClassVersions)
+
+	def result = []
+	if (encapCmd) {
+		result += zwaveEvent(encapCmd)
+	}
+	else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+	}
+	return result
+}
+
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv1.WakeUpNotification cmd) {
+	logDebug "Device Woke Up"
+
+	def cmds = []
+	if (state.refreshConfig || pendingChanges > 0) {
+		cmds += getConfigCmds()
+	}
+
+	if (canReportBattery()) {
+		cmds << batteryGetCmd()
+	}
+
+	if (state.refreshSensors) {
+		cmds += [
+				sensorBinaryGetCmd(),
+				sensorMultilevelGetCmd(tempSensorType),
```

Comment:
If You change minTemperatureOffsetParam or minHumidityOffsetParam and device wakes up then  sensorMultilevelGetCmd will sent to the device twice because getCofingCmds will return that command too.